app/models/casebook/resource.rb

```diff
@@ -0,0 +1,32 @@
+# == Schema Information
+#
+# Table name: casebooks
+#
+#  id            :integer          not null, primary key
+#  title         :string           default("Untitled casebook"), not null
+#  slug          :string
+#  subtitle      :string
+#  headnote      :text
+#  public        :boolean          default(TRUE), not null
+#  root_id       :integer
+#  ordinals      :integer          default([]), not null, is an Array
+#  copy_of_id    :integer
+#  is_alias      :boolean
+#  material_type :string
+#  material_id   :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+class Casebook::Resource < Casebook::Contents
+  default_scope {where.not(material_id: nil)}
+
+  belongs_to :material, polymorphic: true, inverse_of: :casebooks, required: true
```

Comment:
Sure. Casebook::Contents is an abstract class for everything that can be content— so anything that's not a Book, ie Sections and Resources.

A resource is a _reference_ to a piece of real material, like a Case or a Text. So it has to be polymorphic to represent the different things it could point to.

Think of the Casebook::Resource as the table of contents entry, where the `material` is the actual content of that section. (This column could be renamed `resource`.)